drivers/watchdog/wdt_dw.c

```diff
@@ -96,10 +96,22 @@ static int dw_wdt_install_timeout(const struct device *dev, const struct wdt_tim
 		return -ENOTSUP;
 	}
 
+#if CONFIG_WDT_DW_RESET_NONE
 	if (config->flags) {
 		LOG_ERR("Watchdog behavior is not configurable.");
 		return -ENOTSUP;
 	}
+#elif CONFIG_WDT_DW_RESET_CPU
+	if (config->flags != WDT_FLAG_RESET_CPU_CORE) {
+		LOG_ERR("only cpu core reset supported");
+		return -ENOTSUP;
+	}
+#elif CONFIG_WDT_DW_RESET_SOC
+	if (config->flags != WDT_FLAG_RESET_SOC) {
+		LOG_ERR("only SoC reset supported");
+		return -ENOTSUP;
+	}
+#endif
```

Comment:
I agree there is no IP configuration that has to be done by the driver for resetting and it is completely hardware platform design. In agilex5 platform, board supports a system reset through watchdog expiry. Any technical documentation of the board will say that this supports SoC reset or CPU or none of resets based on the board design.

The user may pass the `struct wdt_timeout_cfg` to `wdt_install_timeout(`) function(for eg sample application passes WDT_FLAG_SOC_RESET_SOC). User can pass the `WDT_FLAG_RESET_SOC `flag and the watchdog driver will fail to install the timeout by responding that this is not supported.

I can think of two solution
1. don't check this flag when `wdt_install_timeout()` is called or 
2. use these three different config flags, so that driver will respond to the user properly as he knows what to configure while installing timeout

Reply:
Personally, I would ignore the flag or tell the user to not use it because it is not configurable by the software.

Reply:
Ignored this with adding warning logs to let the user know that this option is not cofigurable through application